src/tools/storage-key-recipe-resolver.ts

```diff
@@ -128,25 +127,24 @@ export class StorageKeyRecipeResolver {
    * @param recipe long-running or ephemeral recipe
    */
   validateHandles(recipe: Recipe) {
-    recipe.handles
-      .filter(h => h.fate === 'map' || h.fate === 'copy')
-      .forEach(handle => {
-        const matches = this.runtime.context.findHandlesById(handle.id)
-          .filter(h => h.fate === 'create');
-
-        if (matches.length === 0) {
-          throw new StorageKeyRecipeResolverError(`No matching handles found for ${handle.localName}.`);
-        } else if (matches.length > 1) {
-          throw new StorageKeyRecipeResolverError(`More than one handle found for ${handle.localName}.`);
-        }
-
-        const match = matches[0];
-        if (!isLongRunning(match.recipe)) {
-          throw new StorageKeyRecipeResolverError(
-            `Handle ${handle.localName} mapped to ephemeral handle '${match.id}'.`
-          );
-        }
-      });
+    const isMapOrCopyFate = (handle: Handle) => handle.fate === 'map' || handle.fate === 'copy';
+    for (const handle of recipe.handles.filter(isMapOrCopyFate)) {
```

Comment:
It's a little strange defining the lambda and then using it in just the one place immediately below. I'd suggest this instead:

```
const mapOrCopyHandles = recipe.handles.filter(
    handle => handle.fate === 'map' || handle.fate === 'copy');
for (const handle of mapOrCopyHandles) { ... }
```

Reply:
I'm happy with that. I was trying to get the whitespace right.